_posts/2019-10-01-intro-to-hmc/hmc-intro.Rmd

```diff
@@ -0,0 +1,283 @@
+
+In any case, physical analogies make for the best start.
+
+## Physical analogies
+
+The classic physical analogy is given in the reference article, Radford Neal's "MCMC using Hamiltonian dynamics" [@2012arXiv1206.1901N], and nicely explained in a [video by Ben Lambert](https://www.youtube.com/watch?v=a-wydhEuAm0).
+
+So there's this "thing" we want to maximize, the loglikelihood of the data under the model parameters. Alternatively we can say, we want to minimize the negative loglikelihood (like loss in a neural network). This "thing" to be optimized can then be visualized as an object sliding over a landscape with hills and valleys, and like with gradient descent in deep learning, we want it to end up deep down in some valley. 
+
+In Neal's own words
+
+
+> In two dimensions, we can visualize the dynamics as that of a frictionless puck that slides over a surface of varying height. The state of this system consists of the position of the puck, given by a 2D vector q, and the momentum of the puck (its mass times its velocity), given by a 2D vector p.
+
+Now when you hear "momentum" (and given that I've primed you to think of deep learning) you may feel that sounds familiar, but even though the respective analogies are related the association does not help that much. In deep learning, momentum is commonly praised for its avoidance of ineffective oscillations in imbalanced optimization landscapes ^[Though see [this great distill.pub post](https://distill.pub/2017/momentum/) for a different view.].
+With HMC however, the focus is on the concept of _energy_.
+
+In [statistical mechanics](https://en.wikipedia.org/wiki/Statistical_mechanics), the probability of being in some state is inverse-exponentially related to its energy: ^[Here $T$ is the _temperature_; we won't focus on this so just imagine it being set to 1 in this and subsequent equations.]
+
+
+  $$P(E_i) \sim e^{\frac{-E_i}{T}} $$
+
+As you might or might not remember from school physics, energy comes in two forms: potential energy and kinetic energy. In the sliding-object scenario, the object's potential energy corresponds to its height (position), while its kinetic energy is related to its momentum by the formula ^[$m$ here stands for momentum.]
```

Comment:
Also is writing out 'mass' standard notation?

Reply:
I don't think so, probably not... I was trying to exactly follow the notation(s) of my sources, (1) the Ben Lambert video and (2) Radford Neal's article.
I'm also trying to say that explicitly (in case someone wonders why the notation changes).
In the beginning, I was thinking to "unify" (wrong word but you know what I mean) it all, but in the end it (1) seemed safest and (2) should be nice for readers who go on to watch/read those original sources directly...
Does this make sense?

Reply:
yep if that's what the references use it's fine

Reply:
that's good to hear!